### PR TITLE
Fix idle connection disposal

### PR DESCRIPTION
**Pull request by jaceksokol**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://github.com/ReactiveMongo/ReactiveMongo/blob/master/CONTRIBUTING.md#reactivemongo-developer--contributor-guidelines)?
* [ ] Have you added tests for any changed functionality?

## Fixes

Fixes an issue described in: https://groups.google.com/g/reactivemongo/c/3Kb-tpAcO6Y

## Purpose

Fixes idle connection disposal.

## Background Context

The `IdleStateHandler` should be set as first in the netty's pipeline (according to the documentation). This was not true if SSL was enabled.

## References

* Described issue with code to reproduce: https://groups.google.com/g/reactivemongo/c/3Kb-tpAcO6Y
* Netty's docs: https://netty.io/4.0/api/io/netty/handler/timeout/IdleStateHandler.html

### CODE REVIEW

File: driver/src/main/scala/core/protocol/MongoHandler.scala [hunk 1]
```diff
@@ -1,51 +1,39 @@
...
-  idleTimeMS: Long) extends IdleStateHandler(
-  idleTimeMS, idleTimeMS, idleTimeMS, TimeUnit.MILLISECONDS) {
+  receiver: ActorRef) extends ChannelDuplexHandler {
 
   private var last: Long = -1L // in nano-precision
 
   override def channelActive(ctx: ChannelHandlerContext): Unit = {
     log(ctx, "Channel is active")
 
-    last = System.nanoTime()
```
Comment (cchantep) [inline driver/src/main/scala/core/protocol/MongoHandler.scala:32]: `last` would never be updated

File: driver/src/main/scala/core/protocol/MongoHandler.scala [hunk 2]
```diff
@@ -1,51 +1,39 @@
 package reactivemongo.core.protocol
 
-import java.util.concurrent.TimeUnit
-
 import akka.actor.ActorRef
-
-import reactivemongo.io.netty.channel.{
```
Comment (cchantep) [inline driver/src/main/scala/core/protocol/MongoHandler.scala:7]: Import reformating not acceptable

File: driver/src/main/scala/core/netty/ChannelFactory.scala [hunk 3]
```diff
@@ -1,39 +1,22 @@
 package reactivemongo.core.netty
 
 import java.lang.{ Boolean => JBool }
-
+import java.util.concurrent.TimeUnit
```
[the remark this answers is not in the document]
Comment (jaceksokol) [reply driver/src/main/scala/core/netty/ChannelFactory.scala:5]: Right, automatically performed by IDE, reverted.

File: driver/src/main/scala/core/protocol/MongoHandler.scala (hunk 1 above)
Comment (jaceksokol) [reply driver/src/main/scala/core/protocol/MongoHandler.scala]: Removed by mistake, reverted.

File: driver/src/main/scala/core/protocol/MongoHandler.scala (hunk 2 above)
Comment (jaceksokol) [reply driver/src/main/scala/core/protocol/MongoHandler.scala]: Reverted.

File: driver/src/main/scala/core/netty/ChannelFactory.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (cchantep) [reply driver/src/main/scala/core/netty/ChannelFactory.scala:5]: Still not reverted, diff visible there

File: driver/src/main/scala/core/protocol/MongoHandler.scala [hunk 4]
```diff
@@ -1,21 +1,36 @@
 package reactivemongo.core.protocol
 
+import java.util.concurrent.TimeUnit
```
Comment (cchantep) [inline driver/src/main/scala/core/protocol/MongoHandler.scala]: Bad revert I guess

**Review by cchantep — CHANGES_REQUESTED** [review]

File: driver/src/main/scala/core/netty/ChannelFactory.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaceksokol) [reply driver/src/main/scala/core/netty/ChannelFactory.scala:5]: This import is now required.

File: driver/src/main/scala/core/protocol/MongoHandler.scala (hunk 4 above)
Comment (jaceksokol) [reply driver/src/main/scala/core/protocol/MongoHandler.scala]: Yes indeed.

**Comment by cchantep** [comment]
Published in 1.0.4

**Pull request merged**